### PR TITLE
cleanup JSR stuff

### PR DESCRIPTION
**Pull request by williamhorning**
this PR should help cleanup the JSR score by automating publishing, adding docs, and fixing imports

additionally, this also bumps the dependency versions and makes all the tests pass (at least on my end)

### CODE REVIEW

**Review by lucsoft — APPROVED** [review]
LGTM :shipit:

File: .github/FUNDING.yml
Comment (erfanium) [inline .github/FUNDING.yml:1]: fell free to remove this. I'm not open anymore to do this

Comment (lucsoft) [reply .github/FUNDING.yml:1]: You change it to my name I'm still looking forward adding more features

Comment (williamhorning) [reply .github/FUNDING.yml:1]: just updated the file to point to @lucsoft

**Comment by williamhorning** [comment]
the test failure on the latest commit seems to just be a fluke since running the tests [again] 
(https://github.com/williamhorning/mongo/actions/runs/9127552219/job/25098162274) causes the failing test to pass

**Pull request merged**